issue: useController causing type error from 7.63.0 -> 7.64.0

### Version Number

7.64.0

### Codesandbox/Expo snack

https://codesandbox.io/p/devbox/lucid-frost-gnfyfx

### Steps to reproduce

1. Run `tsc`
2. Receive error:
```
app/components/tree/tree.tsx:37:37 - error TS2322: Type 'string' is not assignable to type 'never'.

37   const { field } = useController({ name, defaultValue: [] });
                                       ~~~~

  node_modules/react-hook-form/dist/types/controller.d.ts:20:5
    20     name: TName;
           ~~~~
    The expected type comes from property 'name' which is declared here on type 'UseControllerProps<never[], never, never[]>'

app/components/tree/tree.tsx:37:43 - error TS2322: Type 'never[]' is not assignable to type 'undefined'.

37   const { field } = useController({ name, defaultValue: [] });
                                             ~~~~~~~~~~~~

  node_modules/react-hook-form/dist/types/controller.d.ts:23:5
    23     defaultValue?: FieldPathValue<TFieldValues, TName>;
           ~~~~~~~~~~~~
    The expected type comes from property 'defaultValue' which is declared here on type 'UseControllerProps<never[], never, never[]>'

app/components/tree/tree.tsx:46:26 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'never'.

46       if (value.includes(itemId)) {
                            ~~~~~~

app/components/tree/tree.tsx:47:9 - error TS2322: Type 'string[]' is not assignable to type 'never[]'.
  Type 'string' is not assignable to type 'never'.

47         value = without(value, itemId);
           ~~~~~

app/components/tree/tree.tsx:49:20 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'never'.

49         value.push(itemId);
```

### Expected behaviour

Expect no type errors on minor upgrade.

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/src/__typetest__/useController.test-d.ts
+++ b/src/__typetest__/useController.test-d.ts
@@ -1,0 +1,85 @@
+import { expectNotType, expectType } from 'tsd';
+
+import { useController } from '../useController';
+
+/** {@link useController} */ {
+  /** it should NOT infer never[] when defaultValue is empty array without explicit types */ {
+    /* eslint-disable react-hooks/rules-of-hooks */
+
+    const { field } = useController({
+      name: 'items' as string,
+      defaultValue: [],
+    });
+
+    expectNotType<never>(field.name);
+    expectNotType<never[]>(field.value);
+
+    if (Array.isArray(field.value)) {
+      field.value.includes('test');
+      field.value.push('test');
+    }
+  }
+
+  /** it should work correctly with explicit type parameters */ {
+    /* eslint-disable react-hooks/rules-of-hooks */
+
+    type FormValues = {
+      items: string[];
+    };
+
+    const { field } = useController<FormValues, 'items'>({
+      name: 'items',
+      defaultValue: [],
+    });
+
+    expectType<string[]>(field.value);
+    expectType<'items'>(field.name);
+  }
+
+  /** it should work with optional array fields */ {
+    /* eslint-disable react-hooks/rules-of-hooks */
+
+    type FormValues = {
+      optionalItems?: string[];
+    };
+
+    const { field } = useController<FormValues, 'optionalItems'>({
+      name: 'optionalItems',
+      defaultValue: undefined,
+    });
+
+    expectType<string[] | undefined>(field.value);
+  }
+
+  /** it should work with nested array access */ {
+    /* eslint-disable react-hooks/rules-of-hooks */
+
+    type FormValues = {
+      nested: {
+        items: number[];
+      };
+    };
+
+    const { field } = useController<FormValues, 'nested.items'>({
+      name: 'nested.items',
+      defaultValue: [],
+    });
+
+    expectType<number[]>(field.value);
+  }
+
+  /** it should handle array of objects */ {
+    /* eslint-disable react-hooks/rules-of-hooks */
+
+    type FormValues = {
+      users: Array<{ id: string; name: string }>;
+    };
+
+    const { field } = useController<FormValues, 'users'>({
+      name: 'users',
+      defaultValue: [],
+    });
+
+    expectType<Array<{ id: string; name: string }>>(field.value);
+  }
+}

--- a/src/types/path/eager.ts
+++ b/src/types/path/eager.ts
@@ -165,7 +165,7 @@ type PathValueImpl<T, P extends string> = T extends any
       : P extends `${ArrayKey}`
         ? T extends ReadonlyArray<infer V>
           ? V
-          : undefined extends T
+          : T extends undefined
             ? undefined
             : never
         : never

